### PR TITLE
RssAgent: Add limit on feed items fetched

### PR DESCRIPTION
**Pull request by irfancharania**
How's this for a start?
I'm going to look at `rss_agent_spec` file later and try to figure out how to add a test in there

### CODE REVIEW

File: app/models/agents/rss_agent.rb [hunk 1]
```diff
@@ -76,8 +77,12 @@ def check
         if response.success?
           feed = FeedNormalizer::FeedNormalizer.parse(response.body)
           feed.clean! if interpolated['clean'] == 'true'
+
+          max_events = Integer(interpolated['max_items_per_feed']) if options['max_items_per_feed'].present?
+
           created_event_count = 0
           feed.entries.sort_by { |entry| [entry.date_published, entry.last_updated] }.each do |entry|
+            break if (!max_events.nil?) && (max_events >= 0) && (created_event_count >= max_events)
```
Comment (cantino) [inline app/models/agents/rss_agent.rb]: I might do

```ruby
max_events = (interpolated['max_items_per_feed'].presence || 0).to_i
feed.entries.sort_by { |entry| [entry.date_published, entry.last_updated] }.each.with_index do |entry, index|
  break if max_events && max_events > 0 && index >= max_events
  ...
```

**Comment by cantino** [comment]
Nice!

**Comment by knu** [comment]
Does it set a limit on how many events can be created per run?  If so, I would call it "max_events_per_run" and probably make it a common option in the future.

**Comment by irfancharania** [comment]
OK, sounds good

**Comment by irfancharania** [comment]
Related to issue #858

**Comment by cantino** [comment]
Looks good!  If this is working for you, I'll merge it.

**Comment by irfancharania** [comment]
Yep, it's working for me

**Comment by cantino** [comment]
Thanks @irfancharania, great work!

Would you mind sending me a quick email at andrew at andrewcantino dot com?

**Pull request merged**